pandas/core/groupby/generic.py

```diff
@@ -1731,7 +1731,11 @@ def _wrap_aggregated_output(
         DataFrame
         """
         indexed_output = {key.position: val for key, val in output.items()}
-        columns = Index(key.label for key in output)
+        if self.axis == 0:
```

Comment:
I think this can just be `self._obj_with_exclusions._get_axis_name(self.axis)`

Reply:
Awesome, that is much better. Thanks!

Reply:
I spoke to soon. This method returns either "index" or "columns", not the name that we desire here.

Reply:
Ah ha - found it! `self._obj_with_exclusions._get_axis_name(1-self.axis).name`